db/mongo/mongoClient.go

```diff
@@ -0,0 +1,124 @@
+package mongo
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	guuid "github.com/google/uuid"
+	"github.com/spf13/viper"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/abdulmoeid7112/student-backend-portal/config"
+	"github.com/abdulmoeid7112/student-backend-portal/db"
+	"github.com/abdulmoeid7112/student-backend-portal/models"
+)
+
+//ErrorFooFailedError ...
+var ErrorFooFailedError = errors.New("failed to connect")
+
+//ErrorFooFailedAdd ...
+var ErrorFooFailedAdd = errors.New("failed to add")
+
+//ErrorFooIDEmpty ...
+var ErrorFooIDEmpty = errors.New("id is not empty")
+
+//ErrorFooFailedUpdate ...
+var ErrorFooFailedUpdate = errors.New("failed to update")
+
+//ErrorFooFailedFound ...
+var ErrorFooFailedFound = errors.New("user not found")
+
+//ErrorFooFailedDelete ...
+var ErrorFooFailedDelete = errors.New("failed to add")
+
+//ErrorFooFailedRetrieve ...
+var ErrorFooFailedRetrieve = errors.New("failed to retrieve")
+
+//ErrorFooFailedRead ...
+var ErrorFooFailedRead = errors.New("failed to read records")
+
+const (
+	collectionName = "Student"
+)
+
+func init() {
+	db.Register("mongo", NewMongoFactory)
+}
+
+type mongoStd struct {
+	conn *mongo.Client
+}
+
+func (m mongoStd) AddStudent(student *models.Student) (string, error) {
+	if student.ID != "" {
+		return "", ErrorFooIDEmpty
+	}
+	student.ID = guuid.New().String()
+	collection := m.conn.Database(viper.GetString(config.DbName)).Collection(collectionName)
+	if _, err := collection.InsertOne(context.TODO(), student); err != nil {
+		return "", err
+	}
+	return student.ID, nil
+}
+
+func (m mongoStd) UpdateStudent(student *models.Student) error {
+	collection := m.conn.Database(viper.GetString(config.DbName)).Collection(collectionName)
+	if _, err := collection.UpdateOne(context.TODO(), bson.M{"_id": student.ID}, bson.M{"$set": student}); err != nil {
+		return ErrorFooFailedUpdate
+	}
+	return nil
+}
+
+func (m mongoStd) GetStudentByID(id string) (*models.Student, error) {
+	var stu *models.Student
+	collection := m.conn.Database(viper.GetString(config.DbName)).Collection(collectionName)
+	if err := collection.FindOne(context.TODO(), bson.M{"_id": id}).Decode(&stu); err != nil {
+		if err == mongo.ErrNoDocuments {
+			return nil, ErrorFooFailedFound
+		}
+		return nil, err
+	}
+	return stu, nil
+}
+
+func (m mongoStd) ListStudent() ([]*models.Student, error) {
```

Comment:
Please refer to this comment: https://github.com/abdulmoeid7112/student-backend-portal/pull/5/files#r448780581